.github/workflows/macos-ci-build-and-test-workflow.yml

```diff
@@ -48,9 +48,9 @@ jobs:
         #
         include: ${{ fromJSON(inputs.matrix_include) }}
 
-    # "macos-13" is a x86_64 image, and "macos-15" is an arm64 image.
+    # macos-15 supports both x86_64 and arm64
```

Comment:
I thought that "macos-15" is an arm64 image.

https://github.com/actions/runner-images/blob/cbcbee08ee2add31e9696c883c2522f06f07c298/README.md?plain=1#L27

will we actually get an Intel Mac if `matrix.machine == 'x86_64'`?

Reply:
It was caused by AI hallucination.  No, we won't have any x86-64 macOS machines from GH.